src/client/app/effects/snippet.ts

```diff
@@ -34,6 +34,7 @@ export class SnippetEffects {
                 .map((snippet: ISnippet) => ({ snippet, mode }))
                 .filter(({ snippet }) => !(snippet == null))
                 .mergeMap(({ snippet, mode }) => this._massageSnippet(snippet, mode, isViewMode))
+                .mergeMap(actions => (actions))
```

Comment:
massageSnippet now returns a Promise of an Observable (Promise<Observable<>>) because of the async/await code, which needs to be just an Observable for the purposes of the effect, hence the extra step.